servatrice/src/serversocketinterface.cpp

```diff
@@ -946,14 +960,21 @@ Response::ResponseCode AbstractServerSocketInterface::cmdRegisterAccount(const C
         {
             QSqlQuery *query = sqlInterface->prepareQuery("insert into {prefix}_activation_emails (name) values(:name)");
             query->bindValue(":name", userName);
-            if (!sqlInterface->execSqlQuery(query))
+            if (!sqlInterface->execSqlQuery(query)) {
+                rc.setResponseExtension(re);
                 return Response::RespRegistrationFailed;
+            }
 
+            re->set_custom_reason_str(servatrice->getCustomRegSuccessMessage().toStdString());
+            rc.setResponseExtension(re);
```

Comment:
Also the custom_reason_str gets set either as a success or failure returning the appropriate string in the ini.  The server operator can define a "success" string and a "failure" string to return.